### PR TITLE
Special syntax to disable tag name checking

### PR DESCRIPTION
**Pull request by glmars**
Introducing a special syntax to disable tag name checking (https://github.com/ThoughtWorksInc/Binding.scala/issues/43#issuecomment-281709586):
```scala
@dom def tag = <raw:custom-tag/>
```
Unfortunately, it is not possible to use the same syntax as for attributes, because `data` name is already busy (there is `data` html tag).

I have selected a `raw:` prefix (similar to `raw"""   """` string interpolator :) )

### CODE REVIEW

**Comment by Atry** [comment]
Is it possible to override `Tags.data` to `override lazy val data = new TypedTag[Element] with Dynamic { ... }`?

File: dom/src/main/scala/com/thoughtworks/binding/dom.scala [hunk 1]
```diff
@@ -405,6 +414,18 @@ object dom {
...
         }
 
+        private def propertyAccess(xmlName: QName, objectAccess: RefTree): Select = {
+          xmlName match {
+            case UnprefixedName(localPart) =>
+              q"$objectAccess.${TermName(NameTransformer.encode(localPart))}"
+            case PrefixedName(prefix, localPart) =>
+              localPart.split(':').foldLeft(q"$objectAccess.${TermName(NameTransformer.encode(prefix))}") {
+                (prefixExpr, segmentName) =>
+                  q"$prefixExpr.${TermName(NameTransformer.encode(segmentName))}"
```
Comment (Atry) [inline dom/src/main/scala/com/thoughtworks/binding/dom.scala]: This change enables `svg` tags by simply adding an additional `val svg: scalatags.jsdom.SvgTags`.

**Comment by Atry** [comment]
I merged this PR with a little change (using `data` prefix instead of `raw`)

@glmars Thank you! You introduced so many wonderful features in Binding.scala.

File: dom/src/main/scala/com/thoughtworks/binding/dom.scala [hunk 2]
```diff
@@ -144,14 +146,16 @@ object dom {
...
+          extends TypedTag[HTMLElement]("data", Nil, false, Namespace.htmlNamespaceConfig)
+          with Dynamic {
         final def selectDynamic(tagName: String): ConcreteHtmlTag[Element] = {
-          tag(tagName)
+          TagsAndTags2.tag(tagName)
         }
-
       }
 
+      override lazy val data = new DynamicDataTag()
```
Comment (glmars) [inline dom/src/main/scala/com/thoughtworks/binding/dom.scala]: A good hack :+1: 
If you like it, I like too :smile:

**Comment by glmars** [comment]
Thank you for the excellent library!
I'm glad to help.

**Pull request merged**